mono/metadata/sgen-mono.c

```diff
@@ -1816,8 +1840,8 @@ mono_gc_alloc_string (MonoVTable *vtable, size_t size, gint32 len)
 	UNLOCK_GC;
 
  done:
-	if (G_UNLIKELY (mono_profiler_events & MONO_PROFILE_ALLOCATIONS))
-		mono_profiler_allocation (&str->object);
+	if (G_UNLIKELY (mono_profiler_allocations_enabled ()))
```

Comment:
I think that would be confusing behavior. In that case, if a profiler installs an allocation callback but doesn't call `mono_profiler_enable_allocations`, it'd be notified of allocations from C code, but not from those done in managed allocators. It seems better to require a profiler to call `mono_profiler_enable_allocations` if it wants allocation notifications *at all*.